modules/ROOT/pages/whats-new-in-dw.adoc

```diff
@@ -1,64 +1,110 @@
-* xref:dw-math.adoc[dw::util::Math] is a new DataWeave module with functions and variables that provides mathematical functions, for example, the function `sin`.
-* xref:dw-periods.adoc[dw::util::Periods] is a new DataWeave module with functions and types to simplify the creation of Periods, for example, the function `days`.
-* xref:dw-tree.adoc[dw::util::Tree] introduces new functions:
-  ** xref:dw-tree-functions-filterarrayleafs.adoc[filterArrayLeafs]
-  ** xref:dw-tree-functions-filterobjectleafs.adoc[filterObjectLeafs]
-  ** xref:dw-tree-functions-filtertree.adoc[filterTree]
-  ** xref:dw-tree-functions-isarraytype.adoc[isArrayType]
-  ** xref:dw-tree-functions-isattributetype.adoc[isAttributeType]
-  ** xref:dw-tree-functions-isobjecttype.adoc[isObjectType]
-* xref:dw-values-types.adoc[dw::util::Values] provides a new type `UpdaterValueProvider`.
+DataWeave 2.5 is bundled with Mule 4.5. The 2.5 version of DataWeave introduces some new features.
+
+== Versioning Behavior in DataWeave
+
+DataWeave now enables backward compatibility with previous 2.x versions of DataWeave at the Mule application and DataWeave script levels:
+
+* Compatibility flags are a set of DataWeave xref:dataweave-system-properties.adoc[DataWeave system properties] that enable you to retain specific DataWeave behavior at the Mule application level. The flags are valid for specified DataWeave versions and expire in a future version of DataWeave. Flags change behavior based on the value of the Mule `minMuleVersion` system property, which corresponds to a DataWeave version. See xref:feature-flagging.adoc[] for details.
+//2.5 DOC MERGED: https://github.com/mulesoft/docs-dataweave/pull/214/files
+
+* Syntax-level support is available at the script level through the `%dw` directive (see xref:dataweave-language-introduction.adoc#dw_header[DataWeave Scripts]). Setting the directive to an earlier version of DataWeave avoids any syntax-breaking changes when the DataWeave runtime engine runs the script. This script-level setting enables you to retain earlier behavior in some scripts while using the latest behavior in others.
+// 2.5 DOC MERGED: https://github.com/mulesoft/docs-dataweave/pull/214/files
+
+For detail, see xref:dataweave-versioning-behavior.adoc[].
+
+== Supported Formats
+
+DataWeave now supports the ProtoBuf format and extends its XML support to DTD.
+
+* Protocol Buffers (Protobuf): This format is typically used with gRPC, the Remote Procedure Call (RPC) framework. See xref:dataweave-formats-protobuf.adoc[].
+
+* DTD for XML: DataWeave can read and create cXML. However, the DTDs remain disabled for security reasons. See xref:dw-xml.adoc[].
+//TODO: Does this related to the parseDtd flag?
+//TODO: How does this relate to compatibility flags?
+//From Lische: https://salesforce-internal.slack.com/archives/C011SNL2469/p1671213207700189
+
```

Comment:
TODO: To read a doctype declarations, set the xref:dataweave-system-properties.adoc[system property] `com.mulesoft.dw.xml_reader.parseDtd`.

Reply:
Also, `DocType` is the DW type for the doctype declaration, and `docType` is the variable containing the value of the doctype declaration in a `DocType` value.

Reply:
new module is Dtd (dw::xml::Dtd)